client/state/themes/utils.js

```diff
@@ -251,3 +251,15 @@ export function getThemeTaxonomySlugs( theme, taxonomy ) {
 	const items = get( theme, [ 'taxonomies', taxonomy ], [] );
 	return items.map( ( { slug } ) => slug );
 }
+
+/**
+ * We currently support two types of theme filtering for recommended themes,
+ * with the future possibility of more in the future, which will require some
+ * further work throughout the code base
+ *
+ * @param {boolean} blockThemes Whether or not to filter on Block Themes.
+ * @returns {string}            The string to filter on in an API request.
+ */
+export function getRecommendedThemesFilter( blockThemes = false ) {
+	return blockThemes ? 'block-templates' : 'auto-loading-homepage';
+}
```

Comment:
Wonder if this is future-proof enough. 🤔 
My concern is that we are tightly coupling the filters to `isUsingSiteEditor`, which might not be flexible enough in the future.
At the same time, I can't think of an equally clear way to handle this across multiple files, unless we just went all-in and used the filters directly instead of the ternary.

What I mean is something like this, in `client/my-sites/themes/recommended-themes.jsx`:
```jsx
const ConnectedRecommendedThemes = connect(
	( state ) => {
		const siteId = getSelectedSiteId( state );
		const isUsingSiteEditor = isSiteUsingCoreSiteEditor( state, siteId );
		const filter = isUsingSiteEditor ? 'block-templates' : 'auto-loading-homepage';
		return {
			recommendedThemes: getRecommendedThemesSelector( state, filter ),
			isLoading: areRecommendedThemesLoading( state, filter ),
			isUsingSiteEditor,
		};
	},
	{ getRecommendedThemes }
)( RecommendedThemes );
```

It would basically be centralized in only one place, as opposed as calling the function everywhere.
Eventually we would still probably need a function to handle multiple cases, but we would only have to handle it once. 🤔 

(This is totally random and untested, btw, so I might very well have missed lots of stuff)

Reply:
Yeah, I'm not entirely happy with this, either. I didn't want the Component to need to know the implementation details of the state, but it would probably be fine.

> Eventually we would still probably need a function to handle multiple cases

Yup, which is why I was putting this through `getRecommendedThemesFilter` - I would change its signature to not merely accept this boolean but some other arg structure. But, since all we currently have are two options, I wound up in the uncomfortable space of making a gesture towards optimizing for future cases while not going all the way. 

But: I don't want to abstract this any further because it's not actually clear that we'll need more types of recommended themes. If anything changes, it might be to reduce it to only the `block-templates` type.

Reply:
Right, what I mean is that by using the filter string directly we might not need to use the convertion function everywhere, but just by the end consumer (the component basically).

Let me try to see if it would work:

client/state/themes/actions/recommended-themes.js
```js
export function getRecommendedThemes( filter ) {
  return async ( dispatch ) => {
    dispatch( { type: RECOMMENDED_THEMES_FETCH, filter } );
    // ...
  };
  // ...
}
```

client/state/themes/selectors/are-recommended-themes-loading.js
```js
export function areRecommendedThemesLoading( state, filter ) {
  return state.themes.recommendedThemes[ filter ]?.isLoading || false;
}
```

client/state/themes/selectors/get-recommended-themes.js
```js
export function getRecommendedThemes( state, filter ) {
  return state.themes.recommendedThemes[ filter ]?.themes || emptyList;
}
```

client/my-sites/themes/recommended-themes.jsx
```js
fetchThemes() {
  const filter = getRecommendedThemesFilter( this.props.isUsingSiteEditor );
  this.props.getRecommendedThemes( filter );
}

// ...

const ConnectedRecommendedThemes = connect(
  ( state ) => {
    const siteId = getSelectedSiteId( state );
    const isUsingSiteEditor = isSiteUsingCoreSiteEditor( state, siteId );
    const filter = getRecommendedThemesFilter( isUsingSiteEditor );
    return {
      recommendedThemes: getRecommendedThemesSelector( state, filter ),
      isLoading: areRecommendedThemesLoading( state, filter ),
      isUsingSiteEditor,
    };
  },
  { getRecommendedThemes }
)( RecommendedThemes );
```

Also, since the `getRecommendedThemesFilter` helper is kinda unavoidably tied to the state (likely even in the future), we might turn into a selector?
This might simplify things even more, I think. 🤔 
```js
export function getRecommendedThemesFilter( state, siteId ) {
  const isUsingSiteEditor = isSiteUsingCoreSiteEditor( state, siteId );
  return isUsingSiteEditor ? 'block-templates' : 'auto-loading-homepage';
}
```

Reply:
Great feedback, thanks for taking the time to walk through this. Turning `getRecommendedThemesFilter` into a selector that way is 😘🤌🏻

Reply:
> Also, since the getRecommendedThemesFilter helper is kinda unavoidably tied to the state (likely even in the future), we might turn into a selector?

Totally agree with this, especially since it's deriving data from state.